### ISSUE TITLE
setindex! or append!

### ISSUE THREAD

**Issue by alha02** [JuliaGPU/CuArrays.jl#18]
Hi Mike, would it be possible to add support for setindex! or append!? So e.g.
```julia
x = cu(randn(10))
x[1:5] .= cu(randn(5))
```
or
```julia
x = cu(randn(5))
append!(x, cu(randn(5)))
```

**Comment by MikeInnes** [comment]
setindex! should be pretty easy (should mostly be able to copy the getindex kernel I think).`append!` is a bit trickier without some extra indirection, and I'm not sure how much it would add over `vcat` anyway (since you still have to allocate space under the hood).

**Comment by MikeInnes** [comment]
Addressed by #29.
